src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +19,55 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY - ampHours_ ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+
+
+    if(previousCurrent_ < 0)
+    {
     return false;
+    }
+    else
+    {
+       return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    double timeRemaining = (BATTERY_AMP_HOUR_CAPACITY-ampHours_) / previousCurrent_ ;
+
+    timeRemaining = qAbs(timeRemaining) / MILISECONDS_TO_HOURS;
+
+    return QTime(0, 0, 0, 0).addMSecs(timeRemaining);
```

Comment:
This whole function could do just being 3 lines, no need to split it up with new lines at all.